addons/mail/controllers/thread.py

```diff
@@ -127,3 +127,12 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
             message_sudo, body, attachment_ids=attachment_ids, partner_ids=partner_ids
         )
         return message_sudo._message_format()[0]
+
+    @http.route("/mail/partner_avatar_card/info", methods=["POST"], type="json", auth="user")
+    def mail_partner_avatar_card_info(self, avatar_id, userFieldNames):
```

Comment:
I don't understand the necessity of this method. Moreover, its implementation looks odd:
- The method receives an ID that could be either a `partner_id` or a `user_id`. Why should we guess? Instead, why not call `search_read` from the web client directly instead of creating a custom route?
- The `userFieldNames` variable should be in snake_case. It is also inconsistent to use this variable in one case and a hard-coded list of partner fields in another.

Reply:
IMO, there are three scenarios to consider:

1. When the received ID is a `user_id` (default case for the avatar card).
2. When a `partner` is mentioned and the partner `has a user_id` (e.g., Marc Demo). 
3. When a `partner` is mentioned but `does not have a user_id` (e.g., Azure Interior).

If we only needed to handle scenarios 1 and 2, we could pass `res_model` as a prop to the `avatar_card` along with `id`, indicating which model to query with an ORM call. However, since scenario 3 is possible, deciding the correct model for the ORM call is not straightforward.

First, we need to check if the partner has a user_id. If they do, we retrieve the data for the mentioned fields. If not, we make another ORM call to get data from the res.partner model. To handle the extra ORM call, I have added a route to retrieve the data from the server. Please let me know if there's a way to do this directly from the web client :)